src/search.js

```diff
@@ -1,31 +1,31 @@
-import {assoc, concat, filter, has, isEmpty, map, pipe, prop, sortBy} from "ramda"
+import {ascend, assoc, concat, descend, has, isEmpty, map, prop, sortBy, sortWith} from "ramda"
 import * as diacritics from 'diacritics'
 
 
 export function findParametersAndVariables(parameters, variables, query) {
-  function matchesQuery(parameterOrVariable) {
-    if (isEmpty(normalizedQuery)) {
-      return true
-    }
-    return parameterOrVariable.name.includes(normalizedQuery) ||
-      parameterOrVariable.searchIndex.includes(normalizedQuery)
+  function weightByPositionIn(string, substring) {
+    const index = string.indexOf(substring)
+    return index === -1 ? Number.MAX_SAFE_INTEGER : index
   }
   const normalizedQuery = diacritics.remove(query.toLowerCase())
-  return pipe(
-    filter(matchesQuery),
-    sortBy(prop('name')),
-  )(concat(
-    map(assoc('type', 'parameter'), parameters),
-    map(assoc('type', 'variable'), variables),
-  ))
+  const parametersAndVariables = concat(parameters, variables)
+  if (isEmpty(normalizedQuery)) {
+    return sortBy(prop('name'), parametersAndVariables)
+  } else {
+    return sortWith([
+      ascend(item => weightByPositionIn(item.name, normalizedQuery)),
+      descend(item => item.normalizedDescription.includes(normalizedQuery)),
+      ascend(prop('name')),
+    ], parametersAndVariables)
+  }
 }
 
-export function withSearchIndex(propertyName, objects) {
+export function withNormalizedDescription(objects) {
```

Comment:
I would recommend to always [use a verb](http://stackoverflow.com/a/1991341) in function names that have side effects. In that case, `withNormalizedDescription` -> `addNormalizedDescription`.